doc/developer/design/20210311_prometheus_source.md

```diff
@@ -0,0 +1,144 @@
+# Summary
+
+This design doc introduces a way to import data from prometheus metrics endpoints into materialized views that can be inspected with SQL commands.
+
+# Goals
+
+* Allow our users to read & make decisions based on our metrics data without setting up a whole metrics ingestion/presentation pipeline.
+
+* Let administrators inspect that metrics data (as much as metrics' attached data allows) together with the other available data we present in system tables.
+
+* Retain a long-enough history of metrics values to allow making decisions based on how the metrics values change (but short enough to not consume unbounded memory).
+
+# Non-Goals
+
+* Export our system table logs as prometheus (yet) -- this design covers only the "metrics import" direction.
+
+* Come up with a data retention scheme other than "older than n minutes gets dropped". If this was a full-featured time series database design, we'd want to thin metrics into minute/hour/day buckets. This isn't that.
+
+* (Maybe) don't even expose the `CREATE SOURCE` mechanics externally?
```

Comment:
Strong agree on this one for this design, I think that's an entirely separate project.